tools/azsdk-cli/Azure.Sdk.Tools.Cli/Microagents/Tools/ListFilesTool.cs

```diff
@@ -0,0 +1,49 @@
+
+using System.ComponentModel;
+
+namespace Azure.Sdk.Tools.Cli.Microagents.Tools;
+
+public record ListFilesInput(
+    [property: Description("The path to list files for, which must be a relative path.")] string Path,
+    [property: Description("Whether to list the files recursively")] bool Recursive,
+    [property: Description("An optional glob pattern to filter the listed files")] string? Filter
+    );
+
+public record ListFilesOutput(IList<string> fileNames);
+
+public class ListFilesTool(string baseDirectory) : AgentTool<ListFilesInput, ListFilesOutput>
+{
+    public override string Name { get; init; } = "ListFiles";
+    public override string Description { get; init; } = "List files in the given directory";
+
+    private readonly string baseDirectory = baseDirectory;
+
+    public override Task<ListFilesOutput> Invoke(ListFilesInput input, CancellationToken ct)
```

Comment:
This method is not async. Any specific reason to return Task instead of `ListFilesOutput`